atlas_metadata_validator/fetch.py

```diff
@@ -62,18 +65,34 @@ def is_valid_url(url, logger=None, retry=10):
         return False
 
 
-def get_controlled_vocabulary(category, resource="translations"):
+def get_controlled_vocabulary(category, resource="atlas", logger=None):
     """Read the json with controlled vocab and return the dict for the given category.
-    The resource parameter specifies which file to read."""
-    resource_package = "atlas_metadata_validator"
+    The config is fetched from the GitHub online copy as the primary source,
+    if this isn't possible ("offline mode") the config packaged with the validator is used.
```

Comment:
I wonder if a regularly updated local cache wouldn't be a better and more consistent way of doing this? If you actually clone the repo with the config file and do a 'git pull' each time, then it will only get updated when it needs to be (when something changes on the remote repo), and you'll have a local cache to fall back to.

Otherwise you're stuck trying to keep the packaged version in synch or risk opaque confusing behaviour in future.

Reply:
e.g. see what the analysis control workflow does with the metadata repo: https://github.com/ebi-gene-expression-group/scxa-control-workflow/blob/8c4fac5ae48e83e9113e734c3b47bd73b9c1644e/bin/submitIndependentControlWorkflows.sh#L75

Reply:
Yes, that was one of the worries, still having two copies of the same file. I like that idea to pull the whole repo every time as it isn't expected to become very big. Let me give that a try (need to figure out the best way to do that in Python).